### ISSUE TITLE
Support unicode input in built-in REPL

### ISSUE THREAD

**Issue by crocket** [janet-lang/janet#284]
For now, unicode input is ignored. I want to input something like

```
> (print "おはよう")
```

**Comment by oliwer** [comment]
I guess this could be done using this version of linenoise: https://github.com/rain-1/linenoise-mob

**Comment by kamisori** [comment]
Are you perhaps using a terminal emulator that doesn't support unicode?
When I start janet in my gitbash I can put in any unicode I want just fine:

![EZlzagcXgAMmC9B](https://user-images.githubusercontent.com/293332/83887990-7d9b4380-a749-11ea-980b-e8d19702664e.png)

**Comment by sogaiu** [comment]
FWIW, I have more luck when I start janet using `-s`.

**Comment by bakpakin** [comment]
This should be fixed in e8734c77b462ec7d4470f0be22a4d6cc6414fc73

**Comment by sogaiu** [comment]
Input seems to have improved:
```
repl:1:> "昇竜拳"
"\xE6\x98\x87\xE7\xAB\x9C\xE6\x8B\xB3"
```
Some bits I noticed:

If I press the up arrow key at the next prompt, the spot the cursor appears at seems shifted to the right a bit.  Specifically, the cursor appears beneath the `B` of `\xAB`.  I think the cursor should actually be in the column that the `7` of `\xE7` is in.

If I then press backspace once, the right-most double quote is deleted even though the cursor appears to be more than one position to the right of the double quote.

A subsequent press of backspace yields:
```
repl:2:> "昇竜�
```
I guess what is going on here has partly to do with each of these non-ascii characters being multibyte but also the displayed character's width is actually twice that of ascii characters.

These bits seems like they'd be a bit of work to adjust for.

I'm happy that at least input seems to work better so far though.

Thanks!
